src/backend/distributed/commands/table.c

```diff
@@ -71,53 +75,97 @@ PreprocessDropTableStmt(Node *node, const char *queryString)
 
 	Assert(dropTableStatement->removeType == OBJECT_TABLE);
 
+	List *citusTableOidList = ExtractCitusTableOidsFromDropStmt(dropTableStatement);
+
+	if (list_length(citusTableOidList) <= 0)
+	{
+		/* no citus tables are to be dropped by DROP command, simply return */
+		return NIL;
+	}
+
+	/* detach partitions of tables in MX nodes if any */
+	DetachPartitionsFromMXNodes(citusTableOidList);
+
+	return NIL;
```

Comment:
It seems that this method always returns NIL? Am I missing something?

Reply:
yes, this is because all our preprocess functions do have the same the return type (utility_hook.c)